### PR TITLE
Fix iOS scrolling outside ScrollView bounds

### LINKED ISSUE #12603: [Bug] ScrollView ScrollToAsync behaves different on Android and iOS
### Description
ScrollToAsync methode of ScrollView shows a different behaviour on Android and iOS.

[Tested with ScrollToPosition.Start parameter]
On Android the scrolling is performed inside the bounds of the ScrollView. (Expected Behaviour)
On iOS the scrolling is performed in every case and the ScrollView bounds are ignored.  

Example: The layout consists of a ScrollView  that holds a StackLayout with four buttons.
When a button is clicked  `await scrollParent.ScrollToAsync(button, ScrollToPosition.Start, true);` is called for the clicked button.

In case there is a big enough screen, the content fits the screen and the ScrollView won't show ScrollBars. In this case the ScrollView can not be scrolled because all content is inside the visible bounds of the ScrollView.  
IMG_0006

In this state clicking a button will lead to:
Android => No scrolling is performed.
iOS => Scrolling is performed. ScrollView interaction will still be disabled and the user can not scroll back.
The next image shows the iOS state after button No.3 was clicked. 
IMG_0007

 
### Steps to Reproduce
See attached demo project
Test was performed on real devices.
[ScrollATest.zip](https://github.com/xamarin/Xamarin.Forms/files/5437344/ScrollATest.zip)
 

### Expected Behavior
Scrolling should always be performed inside the bounds of the ScrollView
or the platforms should at least behave the same. 

### Actual Behavior
On iOS scroll is performed in every case and the ScrollView bounds are ignored.  

### Basic Information

- Version with issue: 4.8.0.1560
- Last known good version:  unknown
- IDE: Visual Studio Community 2019 16.7.6
- Platform Target Frameworks: 
  - iOS:  Target 11.0, SDK 14.X  
  - Android: Android 11.0 (R) 
  - UWP:  not tested

### Reproduction Link
See attached demo project
[ScrollATest.zip](https://github.com/xamarin/Xamarin.Forms/files/5437344/ScrollATest.zip)

### Workaround
Restrict scrolling to the ScrollView bounds for iOS.
```
// WORKAROUND CODE
if (view?.Parent?.Parent is ScrollView scrollParent)
{
    if (Device.RuntimePlatform == Device.iOS && view.Parent is VisualElement parent)
    {
        double overlap = parent.Height - scrollParent.Height;                           // in case overlap is negative there is no need to scroll because everything is visible  
        double yScrollTo = (overlap > view.Y) ? view.Y : overlap;                       // scroll directly to the element or towards the element as far as as overlap allows
        if (yScrollTo > 0)
            await scrollParent.ScrollToAsync(0, yScrollTo, true);
    }
    else
        await scrollParent.ScrollToAsync(view, ScrollToPosition.Start, true);
}
```

### PR DESCRIPTION
**Pull request by AlleSchonWeg**
### Description of Change ###
Perform scrolling inside the bounds of the ScrollView.

### Issues Resolved ### 

- fixes #12603

### API Changes ###

 
 None

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

If scrolling to an element, which is at the end of the list and setting ScrollToPosition to Start or Center the app now bahaves same as android.

### Before/After Screenshots ### 

Before:
image
After:
image

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/Renderers/ScrollViewRenderer.cs [hunk 1]
```diff
@@ -286,6 +286,10 @@ void OnScrollToRequested(object sender, ScrollToRequestedEventArgs e)
 						break;
 				}
 			}
+			if(newOffset.X + ScrollView.Width > ContentSize.Width)
```
Comment (jsuarezruiz) [inline Xamarin.Forms.Platform.iOS/Renderers/ScrollViewRenderer.cs:289]: Could you include a sample and an UITest (tapping a Button and checking a Label value for example)?.

Comment (AlleSchonWeg) [reply Xamarin.Forms.Platform.iOS/Renderers/ScrollViewRenderer.cs:289]: Hi @jsuarezruiz 
i added a test. If its not enough, please modify the test.
Thank you

Comment (jsuarezruiz) [reply Xamarin.Forms.Platform.iOS/Renderers/ScrollViewRenderer.cs:289]: Thanks!. Could you rebase 5.0.0 branch to fix the conflicts?

Comment (AlleSchonWeg) [reply Xamarin.Forms.Platform.iOS/Renderers/ScrollViewRenderer.cs:289]: @jsuarezruiz 
Conflicts fixed.

**Comment by AlleSchonWeg** [comment]
Hi @jfversluis 
I tested the nuget and with this fix scrolling behaves same as android. Is something missing in this pull request?

**Review by jfversluis — APPROVED** [review]

**Comment by jfversluis** [comment]
Nothing besides me merging it!

**Comment by alsed42** [comment]
For me, this change introduces a new bug where calling ScrollView.ScrollToAsync has no effect on iOS anymore. Managed to restore functionality by integrating the ScrollViewRenderer's source into my project, minus the four lines added in this revision.

**Pull request merged**